crates/emitter/src/control_structures.rs

```diff
@@ -299,16 +298,34 @@ impl ControlStructureStack {
         }
     }
 
-    pub fn find_labelled_index(&mut self, label: SourceAtomSetIndex) -> Option<usize> {
-        self.control_stack.iter().position(|control| match control {
-            Control::Label(control) => {
-                if control.name == label {
-                    return true;
+    pub fn find_labelled_control(&mut self, label: SourceAtomSetIndex) -> &mut Control {
+        self.control_stack
+            .iter_mut()
+            .find(|control| match control {
+                Control::Label(control) => {
+                    if control.name == label {
+                        return true;
+                    }
+                    false
                 }
-                false
-            }
-            _ => false,
-        })
+                _ => false,
+            })
+            .expect("there should be a control with this label")
+    }
+
+    pub fn find_labelled_index(&mut self, label: SourceAtomSetIndex) -> usize {
```

Comment:
can you make this private? it's returning untyped index that shouldn't be exposed